Compiler error "Syntax error, ',' expected" on RotateTransform.Angle

### Current behavior

In UWP, this RotateTransform compiles and parses as expected:

```xaml
<RotateTransform Angle="45.4452 19.5142 28.1078" CenterX="2"/>
```

When compiling under WASM, we see the error:
> error CS1003: Syntax error, ',' expected


### Expected behavior

It should compile and run the same way UWP does.

### How to reproduce it (as minimally and precisely as possible)

Minimal repro:
http://playground.platform.uno/#5692c3dd

Original code, generated by Inkscape as "Silverlight XAML":
```xaml

<Viewbox Height="23" Width="20">
    <Canvas xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation" Name="svg11" Width="94" Height="150">
        <Canvas.RenderTransform>
            <TranslateTransform X="0" Y="0"/>
        </Canvas.RenderTransform>
        <Canvas.Resources/>
        <!--Unknown tag: sodipodi:namedview-->
        <!--Unknown tag: mask-->
        <Rectangle xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" Canvas.Left="19.5142" Canvas.Top="28.1078" Width="105.403" Height="27.3852" RadiusX="2" RadiusY="2" Name="rect5" StrokeThickness="7" Stroke="#ffffff">
            <Rectangle.RenderTransform>
                <RotateTransform Angle="45.4452 19.5142 28.1078" CenterX="2"/>
            </Rectangle.RenderTransform>
        </Rectangle>
        <Path xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" Name="path7" StrokeThickness="7" Stroke="#ffffff" Data="M10.8895 108.635L25.4156 94.3329L61.2694 130.748L46.7433 145.05L10.8895 108.635Z"/>
        <Path xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" Name="path9" StrokeThickness="7" Stroke="#ffffff" Data="M33.4484 19.2517L47.9745 4.9496L81.7496 39.2538L67.2235 53.5559L33.4484 19.2517Z"/>
    </Canvas>
</Viewbox>
```

Simplified XAML:
```xaml
<Rectangle Height="500" Width="500" Fill="Red"> 
    <Rectangle.RenderTransform>
        <RotateTransform Angle="45.4452 19.5142 28.1078" />
    </Rectangle.RenderTransform>
</Rectangle>
```

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia, Uno.UI.RemoteControl / Uno.WinUI.RemoteControl

### NuGet package version(s)

Uno.UI 3.10.11
Uno.UI.WebAssembly 3.10.11
Uno.UI.RemoteControl 3.10.11

### Affected platforms

WebAssembly

### IDE

Visual Studio 2019

### IDE version

16.11.4

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

--- a/src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs
+++ b/src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs
@@ -4427,9 +4427,12 @@ namespace Uno.UI.SourceGenerators.XamlGenerator
 				switch (propertyType.SpecialType)
 				{
 					case SpecialType.System_Int32:
+						// UWP ignores everything starting from a space. So `5 6 7` is a valid int value and it's "5".
+						return IgnoreStartingFromFirstSpaceIgnoreLeading(GetMemberValue());
 					case SpecialType.System_Int64:
 					case SpecialType.System_Int16:
 					case SpecialType.System_Byte:
+						// UWP doesn't ignore spaces here.
 						return GetMemberValue();
 					case SpecialType.System_Single:
 					case SpecialType.System_Double:
@@ -6036,7 +6039,32 @@ namespace Uno.UI.SourceGenerators.XamlGenerator
 			// UWP's Xaml parsing ignores curly brackets at beginning and end of a double literal
 			memberValue = memberValue.Trim('{', '}');
 
+			if (isDouble)
+            {
+				// UWP does that for double but not for float.
+				memberValue = IgnoreStartingFromFirstSpaceIgnoreLeading(memberValue);
+			}
+
 			return "{0}{1}".InvariantCultureFormat(memberValue, isDouble ? "d" : "f");
+		}
+
+		private static string IgnoreStartingFromFirstSpaceIgnoreLeading(string value)
+		{
+			var span = value.AsSpan().TrimStart();
+
+			var firstWhitespace = -1;
+			for (int i = 0; i < span.Length; i++)
+			{
+				if (char.IsWhiteSpace(span[i]))
+				{
+					firstWhitespace = i;
+					break;
+				}
+			}
+
+			return firstWhitespace == -1
+				? value
+				: span.Slice(0, firstWhitespace).ToString();
 		}
 
 		private string ValidatePropertyType(INamedTypeSymbol propertyType, XamlMemberDefinition? owner)

--- a/src/SourceGenerators/XamlGenerationTests/Double_IgnoreStartingFromWhitespace.xaml
+++ b/src/SourceGenerators/XamlGenerationTests/Double_IgnoreStartingFromWhitespace.xaml
@@ -1,0 +1,21 @@
+﻿<UserControl x:Class="XamlGenerationTests.Shared.Double_IgnoreStartingFromWhitespace"
+	xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+	xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+	xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+	xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+	mc:Ignorable="d">
+
+    <Grid>
+        <Rectangle Height="500" Width="500" Fill="Red"> 
+            <Rectangle.RenderTransform>
+                <RotateTransform Angle="45.4452 19.5142 28.1078" />
+            </Rectangle.RenderTransform>
+        </Rectangle>
+
+        <Rectangle Height="500" Width="500" Fill="Red"> 
+            <Rectangle.RenderTransform>
+                <RotateTransform Angle="   45.4452 19.5142 28.1078" />
+            </Rectangle.RenderTransform>
+        </Rectangle>
+    </Grid>
+</UserControl>

--- a/src/Uno.UI.Tests/Windows_UI_Xaml_Data/BindingHelperTests/Given_XamlBindingHelper.cs
+++ b/src/Uno.UI.Tests/Windows_UI_Xaml_Data/BindingHelperTests/Given_XamlBindingHelper.cs
@@ -69,5 +69,34 @@ namespace Uno.UI.Tests.Windows_UI_Xaml_Data.CollectionViewTests
 		{
 			Assert.AreEqual(new System.Drawing.PointF(expectedX, expectedY), XamlBindingHelper.ConvertValue(typeof(System.Drawing.PointF), value));
 		}
+
+		[TestMethod]
+		[DataRow("", 0)]
+		[DataRow("0", 0)]
+		[DataRow("  0  ", 0)]
+		[DataRow("  0  5", 0)]
+		[DataRow("  10  5", 10)]
+		[DataRow("10  5", 10)]
+		[DataRow("10  ", 10)]
+		[DataRow("10", 10)]
+		public void When_ToInt(string value, int expected)
+		{
+			Assert.AreEqual(expected, XamlBindingHelper.ConvertValue(typeof(int), value));
+		}
+
+		[TestMethod]
+		[DataRow("", double.NaN)]
+		[DataRow("0", 0.0)]
+		[DataRow("  0  ", 0.0)]
+		[DataRow("  0  5", 0.0)]
+		[DataRow("  10.0  5", 10.0)]
+		[DataRow("10.0  5", 10.0)]
+		[DataRow("10.0  ", 10.0)]
+		[DataRow("10  ", 10.0)]
+		[DataRow("10.0", 10.0)]
+		public void When_ToDouble(string value, double expected)
+		{
+			Assert.AreEqual(expected, XamlBindingHelper.ConvertValue(typeof(double), value));
+		}
 	}
 }

--- a/src/Uno.UI/DataBinding/BindingPropertyHelper.FastConvert.cs
+++ b/src/Uno.UI/DataBinding/BindingPropertyHelper.FastConvert.cs
@@ -665,9 +665,9 @@ namespace Uno.UI.DataBinding
 					}
 				}
 
-				var trimmed = input.Trim();
+				var trimmed = IgnoreStartingFromFirstSpaceIgnoreLeading(input);
 
-				if (trimmed == "0" || trimmed == "") // Fast path for zero / empty values (means zero in XAML)
+				if (trimmed == "0" || trimmed.Length == 0) // Fast path for zero / empty values (means zero in XAML)
 				{
 					output = 0d;
 					return true;
@@ -781,9 +781,9 @@ namespace Uno.UI.DataBinding
 					}
 				}
 
-				var trimmed = input.Trim();
+				var trimmed = IgnoreStartingFromFirstSpaceIgnoreLeading(input);
 
-				if (trimmed == "0" || trimmed == "") // Fast path for zero / empty values (means zero in XAML)
+				if (trimmed == "0" || trimmed.Length == 0) // Fast path for zero / empty values (means zero in XAML)
 				{
 					output = 0;
 					return true;
@@ -1089,6 +1089,25 @@ namespace Uno.UI.DataBinding
 			}
 
 			return i;
+		}
+
+		private static string IgnoreStartingFromFirstSpaceIgnoreLeading(string value)
+		{
+			var span = value.AsSpan().TrimStart();
+
+			var firstWhitespace = -1;
+			for (int i = 0; i < span.Length; i++)
+			{
+				if (char.IsWhiteSpace(span[i]))
+				{
+					firstWhitespace = i;
+					break;
+				}
+			}
+
+			return firstWhitespace == -1
+				? value
+				: span.Slice(0, firstWhitespace).ToString();
 		}
 	}
 }

